GArDen/model/__init__.py

```diff
@@ -243,8 +247,80 @@ def _extract_targets(self, graphs):
         y = np.ravel(y)
         return y
 
+
 # ------------------------------------------------------------------------------
 
+class OneClassClassifierWrapper(ClassifierWrapper):
+    """Classifier."""
+
+    def __init__(self,
+                 program=SGDClassifier(average=True,
+                                       class_weight='balanced',
+                                       shuffle=True),
+                 nu=0.5):
+        """Construct."""
+        super(OneClassClassifierWrapper, self).__init__(program)
+        self.nu = nu
+
+    def fit(self, graphs):
+        """fit."""
+        try:
+
+            # make matrix
+            data_matrix = vectorize(graphs,
+                                    vectorizer=self.vectorizer,
+                                    **self.params_vectorize)
+            data_matrix_neg = data_matrix.multiply(-1)
+            data_matrix_both = vstack([data_matrix, data_matrix_neg], format="csr")
+            # make labels
+            length = data_matrix.shape[0]
+            y = [-1] * length + [1] * length
+            y = np.ravel(y)
+            # fit:
+            estimator = self.program.fit(data_matrix_both, y)
+            # moving intercept:
+            # this is an obvious possible performance issue
+            score_vector_pairs = [(estimator.decision_function(sparse_vector)[0], sparse_vector)
+                                  for sparse_vector in data_matrix]
+            score_vector_pairs.sort(key=lambda x: x[0])
+            pivot = int(len(score_vector_pairs) * self.nu)
+            estimator.intercept_ -= score_vector_pairs[pivot][0]
+            # calibration:
+            data_matrix_binary = vstack([a[1] for a in score_vector_pairs])
```

Comment:
use a binary maop to identify > threshild elements
use binary vector to extract pos / neg matrix in one shot